client/components/marketing-survey/cancel-purchase-form/index.jsx

```diff
@@ -126,6 +148,49 @@ class CancelPurchaseForm extends React.Component {
 		this.props.onInputChange( newState );
 	};
 
+	getSurveyDataType = () => {
+		if ( this.props.flowType === 'remove' ) {
+			return 'remove';
+		}
+
+		return isRefundable( this.props.purchase ) ? 'refund' : 'cancel-autorenew';
+	};
+
+	onSubmit = () => {
+		const { purchase, selectedSite } = this.props;
+
+		if ( ! isDomainRegistration( purchase ) && ! isGoogleApps( purchase ) ) {
+			this.setState( {
+				isSubmitting: true,
+			} );
+
+			const surveyData = {
+				'why-cancel': {
+					response: this.state.questionOneRadio,
+					text: this.state.questionOneText,
+				},
+				'next-adventure': {
+					response: this.state.questionTwoRadio,
+					text: this.state.questionTwoText,
+				},
+				'what-better': { text: this.state.questionThreeText },
+				type: this.getSurveyDataType(),
+			};
+
+			submitSurvey(
+				'calypso-remove-purchase',
```

Comment:
Is the same name used for both Cancel and Remove flows?

Reply:
Yes. It is simply following the current behavior, as you can see in the following lines:
* https://github.com/Automattic/wp-calypso/blob/683d982/client/me/purchases/remove-purchase/index.jsx#L171
* https://github.com/Automattic/wp-calypso/blob/683d982e5b8fe8ecf5d49f6d74a046362a95f7e9/client/me/purchases/cancel-purchase/button.jsx#L293

However, I should definitely make it different in https://github.com/Automattic/wp-calypso/pull/34492 for the case of turning off auto-renewal, so thanks for the catch!